ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,13 +358,31 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+      checkUnreadyConnectionsStatus();
       this.metrics.selectorIORate.inc();
     }
     long endIo = time.milliseconds();
     this.metrics.selectorIOTime.update(endIo - endSelect);
   }
 
   /**
+   * Check readiness for unready connections and add to completed list if ready
+   */
+  private void checkUnreadyConnectionsStatus() {
+    Iterator<Map.Entry<String, Long>> iterator = connectionPendingHandshakeTimer.entrySet().iterator();
+    while (iterator.hasNext()) {
+      Map.Entry<String, Long> entry = iterator.next();
+      if (isChannelReady(entry.getKey())) {
+        connected.add(entry.getKey());
+        metrics.selectorConnectionCreated.inc();
+        metrics.selectorPerceivedSslHandshakeTime.update(System.currentTimeMillis() - entry.getValue());
```

Comment:
It seems in `SSLTransmission` this handshaking time is already reported.
`metrics.sslHandshakeTime.update(time.milliseconds() - handshakeStartTime);` There may be some difference in value if compared to `selectorPerceivedSslHandshakeTime`, but I guess it would not be so big. Do we still want to keep this?